### ISSUE TITLE
Dedicated Resolver-IP for DNS-Challenge

### ISSUE THREAD

**Issue by markusglaetzner** [tobychui/zoraxy#319]
for those who internaly host an authoritative DNS server (e.g. Technitium DNS), it is not possible to get the acme process working with dns challange. The only solution is to offer an option to enter a dedicated challenge-dns-resolver-ip for the acme-process. Can you add such an option?

**Comment by tobychui** [comment]
@yeungalan Can you take a look at this to see if this is feasible with our current implementation using Lego? Thanks!

**Comment by yeungalan** [comment]
Should be possible, we can take a look if we can implement this

**Comment by markusglaetzner** [comment]
That would be fantastic! When I should test, please let me know.

**Comment by Sickjuicy** [comment]
Hey i have the same problem if you guys need one more test subject :)

**Comment by markusglaetzner** [comment]
Hi, is there any new status on this topic? If there is something to test, let us know. Thanks.

**Comment by tobychui** [comment]
@markusglaetzner No, but I can help you to ping @yeungalan again with no grantee that there will be new status.

**Comment by yeungalan** [comment]
Hey there! Thanks for your interest in this. We're making steady progress on this feature, but since Zoraxy is something we’re working on alongside our full-time jobs, it's moving along a bit slowly. We’ll keep you posted and let you know as soon as it’s ready. Thanks for hanging in there with us!

**Comment by swoop124** [comment]
Hello,

i am also intrested in this. For now, i've configured external dns-servers in docker to verify the certs, but now i can't use internal names anymore.
So it would be very cool, if this feature is implemented.

thanx

**Comment by markusglaetzner** [comment]
In the new version (3.1.2) there still isn't an option to put in a separate ip resolver (e.g. 1.1.1.1) for the let's encrypt process. PollingInterrval and Propagation Timeout are nice options, but they do not fix the dns-problem. Are there any news on this issue?

**Comment by tobychui** [comment]
@markusglaetzner you gonna ask the acme module maintainer for this matter. 

For a quick reminder, this is an open source project developed by contributors in their free time. And from my observations, contributors usually only pick up and prioritize working on things they need or interested. If you really need it urgently, you are welcomed for creating a pull request :)

**Comment by markusglaetzner** [comment]
I was just happy to see that there was an update and since something changed in the Let's Encrypt options, I thought that this option might already be included. Sorry if I came across wrong.

**Comment by Sickjuicy** [comment]
At the weekend, if I have time, I will also see if I can maybe take over the work for the UI part / function

**Comment by markusglaetzner** [comment]
This would help I guess.Von meinem iPhone gesendetAm 25.11.2024 um 20:04 schrieb HazZelnutz ***@***.***>:﻿
I ran into the same problem.
My workaround for now is to hard code Cloudflares DNS server into the DNS01 Provider and recompile.
I made a fork and built a release with this patch, based on the latest 3.1.4 release.
Maybe this helps you for the moment @markusglaetzner.
Keep up the great work!

**Comment by markusglaetzner** [comment]
Great 👍Von meinem iPhone gesendetAm 25.11.2024 um 20:21 schrieb sickjuicy ***@***.***>:﻿
At the weekend, if I have time, I will also see if I can maybe take over the work for the UI part / function

**Comment by Sickjuicy** [comment]
Hey, I did some work over the week and I think I made a working version, but only tested it on my stuff.

https://github.com/Sickjuicy/zoraxy/releases/tag/v3.1.4-DNSnameServer

So pls test it, I made a staging Linux Packed so if you want to test it with the ACME staging server here you go. As Always MAKE A BACKUP OR A CLONE to test

If you need another staging version just say it :) or build it yourself.

**Comment by tobychui** [comment]
@Sickjuicy Nice work! 
When you are ready and comfortable with your version, feel free to create a PR :)

**Comment by leuchtrakete666** [comment]
what a nice coincidence. today i was thinking, i will give zoraxy a shot. i am running an authoritative dns on my network. my first try was to give the docker a dns entry. not perfect at all. 
than i came across this post and the version from @Sickjuicy. i build the docker image and now its just working, without a manual dns entry.

i was testing it with cloudflare. works perfectly. thanks @Sickjuicy for the work.

**Comment by tobychui** [comment]
Interesting, why this issue is still here.
 
![圖片](https://github.com/user-attachments/assets/3b107e35-ae1e-4b8a-a75a-795ee257f506)

It has been added into main branch long time ago (forgotten which version). Not sure why this is not automatically closed. I am closing this now.
